src/platform-integration/windows/extern_win.md

```diff
@@ -0,0 +1,21 @@
+---
+title: External windows in Flutter Windows apps
+description: Special considerations for adding external windows to Flutter apps
+---
+
+## Windows lifecycle
+
+When adding a non-Flutter window to a Flutter Windows app, it will not be part
+of the logic for application lifecycle state updates by default. For example,
+this means that when the external window is shown or hidden, the app lifecycle
+state will not appropriately update to inactive or hidden. In order to add the
+external window to this application logic, the window's `WndProc` procedure
+must invoke `FlutterEngine::ProcessExternalWindowMessage`.
+
+Callbacks can be registered to app lifecycle state changes through
+[WidgetsBindingObserver.didChangeAppLifecycle][].

```

Comment:
I would consider moving this up to the first paragraph where you explain why this migration is necessary.

Perhaps something like:

```
When adding a non-Flutter window to a Flutter Windows app, it will not be part
of the logic for application lifecycle state updates by default. For example,
this means that when the external window is shown or hidden, the app lifecycle
state will not appropriately update to inactive or hidden. As a result,
the app may receive incorrect lifecycle state changes through 
[WidgetsBindingObserver.didChangeAppLifecycle][].

In order to add the
external window to this application logic, the window's `WndProc` procedure
must invoke `FlutterEngine::ProcessExternalWindowMessage`.
```

Reply:
I would be tempted to format it something like this:

# Title

Single line very brief TL;DR style summary of the change.

## Who is affected
All Flutter app developers who maintain a Windows version of their app created that was created using a Flutter version earlier than X.

## What has changed
Details paragraph summarising what changed and how the new world differs from the old.

## What you need to do
Step by step instructions on how to migrate.

## Why this change was necessary
Background on why we needed to make this change.

Reply:
And reading down further... I could have saved myself some writing 🤣